bootstrap/bootstrap.py

```diff
@@ -118,33 +131,39 @@ def verify_sha256(expected_hash: SHA256Hash, f: Path):
     if h != expected_hash:
         raise BadTarball(f, expected_hash, h)
 
+def fetch_file(dest: Path, url: str, sha256: SHA256Hash, network_allowed: bool):
+    if not dest.exists():
+        if not network_allowed:
+            raise MissingSourceFile(url, sha256, dest)
+
+        dest.parent.mkdir(parents=True, exist_ok=True)
+        with urllib.request.urlopen(url) as resp:
+            shutil.copyfileobj(resp, dest.open('wb'))
+
+    verify_sha256(sha256, dest)
+
 def fetch_package(package: PackageName,
                   version: Version,
                   src_sha256: SHA256Hash,
                   revision: Optional[int],
                   cabal_sha256: Optional[SHA256Hash],
+                  network_allowed: bool
                   ) -> (Path, Path):
     import urllib.request
 
     # Download source distribution
+    print(f'Fetching {package}-{version}...')
     tarball = TARBALLS / f'{package}-{version}.tar.gz'
-    if not tarball.exists():
-        print(f'Fetching {package}-{version}...')
-        tarball.parent.mkdir(parents=True, exist_ok=True)
-        url = package_url(package, version)
-        with urllib.request.urlopen(url) as resp:
-            shutil.copyfileobj(resp, tarball.open('wb'))
-
-    verify_sha256(src_sha256, tarball)
```

Comment:
the `verify_sha256` calls seem to be gone. That is not good, they are sanity check.